Makefile

```diff
@@ -329,6 +337,14 @@ version: VERSION mods/ra/mod.yaml mods/cnc/mod.yaml mods/d2k/mod.yaml mods/ts/mo
 		rm $${i}.tmp; \
 	done
 
+cleanversion: VERSION mods/ra/mod.yaml mods/cnc/mod.yaml mods/d2k/mod.yaml mods/ts/mod.yaml mods/modcontent/mod.yaml mods/all/mod.yaml
```

Comment:
This is great.  Can we do it as part of the normal clean?

Reply:
Yes. Updated.

Reply:
Why do we want to set the version when doing a normal clean?

Reply:
Running the packaging scripts dirties the source by setting fixed versions in all the mod.yamls.  Before now we have needed to use git reset to undo that.

Reply:
But the `clean` target can be used outside of packaging so, unless I am missing something, if you do `make && make clean` you will have a dirty tree which is not expected.

Reply:
`Version: {DEV_VERSION}` is the default state for our default mods, so it should leave a clean tree.

Reply:
In practice `make cleanversion` does nothing when you didn't run `make version` before as there is nothing to replace.

Reply:
@abcdefg30 raised the point in IRC that this isn't mirrored in the windows make script, and having to add that reduces my enthusiasm for the idea.

We really shouldn't be running `make version` as part of the packaging scripts, so that we then don't need to clean up after it.  The packaging should be able to set the version itself *on the packaged copy* of the files only.

Reply:
Note that I don't consider mirroring a must though. AFAIK we also didn't do that for `make nunit`.

Reply:
I don't have a Windows development environment set up and I don't know enough about PowerShell so I can't fulfill the request. You can install the Linux Subsystem on Windows nowadays.

Not sure if it is a good idea to hide `make version` in the packaging scripts. It is pretty vital to create distribution packages that are compatible if you clone from git.

Reply:
That could be solved by making it part of the `make install`.

Reply:
I think that would be pretty far of from Makefile conventions.

Reply:
Having the `install` rule make permanent modifications to the source code is a much worse violation of conventions.